packages/Uploader/README.md

```diff
@@ -0,0 +1,95 @@
+# UPLOADER
+
+An agnostic UI component that let you upload files and customize the interface. has a peer dependency on [superagent](https://github.com/visionmedia/superagent)
+
+## Installation
+
+`> npm install --save @paprika/uploader superagent`
+
+or
+
+`> yarn add @paprika/uploader superagent`
+
+## Usage
+
+The `<Uploader />` component uses a render prop pattern expecting a function to pass children.
+
+```js
+<Uploader>
+  {({ ...props }) => {
+    return <YourUI />;
+  }}
+</Uploader>
+```
+
+### The children function
+
+_The children_ function will received the state and helpers that you can use to build your own UI. The following are the properties pass down to the children function.
+
+- **FileInput**
+  The input[type="file"] element ready to be consumed
+- **files**
+  A enhance list of the user selected files
+- **isCompleted**
+  Is true when all files have been processed
+- **isDisabled**
+  Describe the status of the component
+- **isDragLeave**
+  Will be true if a dragOver event occurred and leave the droppable area
+- **isDraggingOver**
+  Is `True` when detecting the user is dragging files over the droppable area
+- **removeFile**
+  `removeFile(key)` will remove a file from the files list, this function doesn't work while the file is on _PROCESSING_ state.
+- **cancelFile**
+  `cancelFile(key)` stop the request cycle keep in mind that if it's on _WAITINGFORSERVER_ state the server might save the file even if - the request has been cancel
+- **upload**
+  `upload()` give you the option to manually upload the files if the consumer decided to not use _hasAutoUpload_
+
+Example:
+
+```js
+<Uploader>
+  {({ FileInput, files, isDisabled, isDraggingOver, isDragLeave, isCompleted, upload, removeFile, cancelFile }) => {
```

Comment:
Couple things:
- I agree it can be useful to have a base component with the functionality that you can customize with the UI. But are there plans for Paprika to implement some "blessed" variant of that UI? Just having a skeleton like this is too bare-bones IMO (pun intended). Imagine everyone consuming this having to implement something that correctly makes use of all the `isDraggingOver`/`isDragLeave`/etc. I guess in other words, is this UI something that should be part of the design system and thus be implemented in Paprika?
- I don't really see the need to surface `FileInput` to the consumer. It _would_ be useful if we were expecting any props on it that the consumer could pass, but it only has children. So it seems like it can just be "inlined" into the uploader component.
- Have you considered an API alternative to a render prop? Seems like we could use context plus a custom hook? I.e. (taking into account the point above):
```jsx
<Uploader>
  <YourStylishUI />
</Uploader>
```
Where `Uploader` provides context and a hook to consume it so that
```jsx
import { useUploader } from '@paprika/uploader';

function YourStylishUI(props) {
  const { files, ... } = useUploader();
  ...
}
```

Reply:
> is this UI something that should be part of the design system and thus be implemented in Paprika
Yes, this PR is only the Component which I'm going to use to implement the UI in my next PR. The developers shouldn't need to use this component, shouldn't even care that exist. 

I didn't add the UI here to keep the PR focus and avoid extra code to review.

About the DS, currently, we have two cases for the FileInput. One the common Drag and Drop, and second when the uploader is a plain button. But of then will benefit from this component. 

> Have you considered an API alternative
No, didn't think about it until reading your comment, I like your idea will change the API. 

> I don't really see the need to surface FileInput 
Is need it because the input file should expand or decrease their size depending the content of the UI, for hover event or focus. 
I will try to see if I can avoid passing it down.